subsys/net/ip/igmp.c

```diff
@@ -195,41 +339,159 @@ static int send_igmp_report(struct net_if *iface,
 	return ret;
 }
 
-enum net_verdict net_ipv4_igmp_input(struct net_pkt *pkt,
-				     struct net_ipv4_hdr *ip_hdr)
+#if defined(CONFIG_NET_IPV4_IGMPV3)
+static int send_igmp_v3_report(struct net_if *iface, struct net_ipv4_igmp_v3_query *igmp_v3_hdr)
 {
-	NET_PKT_DATA_ACCESS_CONTIGUOUS_DEFINE(igmp_access,
-					      struct net_ipv4_igmp_v2_query);
-	struct net_ipv4_igmp_v2_query *igmp_hdr;
+	struct net_if_ipv4 *ipv4 = iface->config.ip.ipv4;
+	struct net_pkt *pkt = NULL;
+	int i, group_count = 0, source_count = 0;
+	const struct in_addr all_systems = ALL_SYSTEMS_ADDR;
+	const struct in_addr igmp_multicast_addr = IGMP_MULTICAST_ADDR;
```

Comment:
Ditto. Perhaps these consts could be even placed at the start of the file instead of placing them in each function.